Templating/Helper/NumberHelper.php

```diff
@@ -183,34 +274,59 @@ protected function getFormatter($culture, $style, $attributes = array(), $textAt
             'style'   => $style,
         ));
 
-        foreach (array_merge($this->textAttributes, $textAttributes) as $name => $value) {
-            $constantName = strtoupper($name);
-            if (!defined('NumberFormatter::'.$constantName)) {
-                throw new \InvalidArgumentException("Numberformatter has no text attribute '$name'");
-            }
-
-            $formatter->setTextAttribute(constant('NumberFormatter::'.$constantName), $value);
+        foreach ($attributes as $attribute => $value) {
+            $formatter->setAttribute($attribute, $value);
         }
 
-        foreach (array_merge($this->attributes, $attributes) as $name => $value) {
-            $constantName = strtoupper($name);
-            if (!defined('NumberFormatter::'.$constantName)) {
-                throw new \InvalidArgumentException("Numberformatter has no attribute '$name'");
-            }
+        foreach ($textAttributes as $attribute => $value) {
+            $formatter->setTextAttribute($attribute, $value);
+        }
 
-            $formatter->setAttribute(constant('NumberFormatter::'.$constantName), $value);
+        foreach ($symbols as $attribute => $value) {
+            $formatter->setSymbol($attribute, $value);
         }
 
         return $formatter;
     }
 
     /**
-     * Returns the canonical name of this helper.
+     * Converts keys of attributes array to values of \NumberFormatter constants.
+     *
+     * @param array $attributes The list of attributes.
      *
-     * @return string The canonical name
+     * @return array List of \NumberFormatter constants.
+     *
+     * @throws \InvalidArgumentException If any attribute does not match any constant.
      */
-    public function getName()
```

Comment:
It will, because only your code change will be shown.

BTW, changing method placing is not the goal of this PR. If you think is a real needed, you can at least propose it on another PR.